setup.py

```diff
@@ -464,6 +772,15 @@ def setup_package():
     except ImportError:
         build_requires = ['numpy>=1.6, <2.0']
 
+    # we require cython because we need to know which part of the wrapper
+    # to build to avoid missing symbols at runtime. But if this script is
+    # called without building pyfftw, for example to install the
+    # dependencies, then we have to hide the cython dependency.
+    try:
+        import cython
+    except ImportError:
+        build_requires.append('cython>=0.23, <1.0')
```

Comment:
Since we are needing Cython anyways, we might consider hardening the requirement on Setuptools and dropping C source from the codebase. The reason being Setuptools (as of [version 18.0]( https://github.com/pypa/setuptools/blob/master/CHANGES.rst#180 ), quite old now) supports compiling the Cython code for us as long as we add Cython to `setup_requires` and point out the Cython source code. Expect that doing this will vastly simplify what we are doing in `setup.py` and simplify our release process (as we are approaching that stage).

Reply:
haha, this has gone round in circles. I think it was @ghisvail who originally advocated including the C source. I honestly don't really care about this (including or not).

Reply:
Interesting. Would be curious to know more about the argument for it (and whether it still applies).

Normally would think that packagers would prefer to be as close to the source of truth as possible. In our case, that is the Cython code. There is always the possibility we forget to generate updated C code. However the same can't be said of the Cython code, which must be up to date and functioning.

Reply:
> Interesting. Would be curious to know more about the argument for it (and whether it still applies).

see pyFFTW/pyFFTW#31 and PyWavelets/pywt#321 for the prior discussion.

Reply:
Also from one of those prior conversations I had found this link to the Cython docs where they recommend distributing the C files:
http://cython.readthedocs.io/en/latest/src/reference/compilation.html#distributing-cython-modules

Reply:
Thanks for the references.

My reading of that issue is that @ghisvail and I would agree.

Have heard people argue for and against the inclusion of C code matching Cython. From my perspective, it's best to keep only one representation of the source around. Since we have made some improvements on the dependency declaration end, expect we will resolve the user issues that prompted inclusion of C code. That said, user problems likely would go away if we just provided binaries.

At the end of the day, don't have particularly strong feelings about this.

Reply:
Since my pr #189 has been merged, we need cython to build because we are now flexible to build if say long double precision is missing. So we shouldn't distribute C source files in the sdist.

Reply:
My bad, I got @ghisvail's position wrong (he's agreeing with you!). I think our current position is reasonable a low cost, as pointed out by @rgommers in https://github.com/PyWavelets/pywt/pull/321.

Reply:
Ach, I just read the whole thread. I can definitely see @fredRos's point. What happens currently if we include it and long double is missing?

Reply:
If long double is missing, different C code is generated from cython by making using of preprocessor-style variables defined in `setup.py`. Here is an example from `pyfftw.pyx`. Note the capital `IF`

```cython
IF HAVE_LONG:
    _supported_types.append('ld')
    _supported_nptypes_complex.append(np.clongdouble)
    _supported_nptypes_real.append(np.longdouble)
```
So you see that we definitely need cython to build

Reply:
Ok, I think I'm convinced as to why we should not include the C file and depend on Cython. There is a clear requirement for explicitly building from the cython source in every case.